### ISSUE TITLE
Migration to Typescript

### ISSUE THREAD

**Issue by imabp** [storyblok/storyblok-js#4]
Migrating to Typescript will help storyblok-js to support for both the languages, Typescript as well as JavaScript.
This also makes sure, storyblok is easily integrate with most common NextJS projects 

We got this idea from PR: #3 and adding @alexjoverm points here:
 
>  - We prefer having the source code in Typescript. In that way, types are autogenerated, better typed, and easier to maintain
>  - Types can be more sophisticated/elaborated
>  - More steps are needed to integrate Typescript (like a few changes in package.json) and we'd rather keep it in 1 PR since is not much work

Would like to work on this issue.

**Comment by imabp** [comment]
👋 Hello @alexjoverm  
I have started working on this and will create a draft PR, once ready with few items in the checklist.

Here is a checklist I have prepared after going through the code.

- [ ] Setup TSConfig and Vite 

- [ ] Plugin Factory Class 
    - This will help to create plugins, by just creating object of the class. 

- [ ] Refactor apiPlugin with Plugin Factory Class

- [ ] Create Interfaces for apiPlugin and use existing StoryblokConfig from `storyblok-js-client`

- [ ] Add custom propertiers to `document.window` for error-free handling of  StoryBlok Bridge

- [ ] Build the package and test it locally.
